### ISSUE TITLE
Installation

### ISSUE THREAD

**Issue by fabianrbz** [fabianrbz/BD2017#2]
#3

**Comment by fabianrbz** [comment]
@HorizonNet @tantalus1984 if we stop aws machines, do we need to setup everything again?
Also, I'm using centos, is there a specific version of the jdk/java/mariadb/connector that I should use? just went with the latest versions. Just asking because the instructions do specify versions for mysql

**Comment by HorizonNet** [comment]
If you everything correctly the services will come up again on their own. You can also restart all the service via Cloudera Manager. If you have a KDC running you probably need to restart the KDC services.

CentOS and RHEL are compatible for our purposes. If you follow the instructions you would go with a recent version of the CentOS 6.x releases. If you are on a CentOS 7.x release you should use a recent version of MariaDB.

My recommendation: stick to the instructions.

**Comment by fabianrbz** [comment]
Thanks! i'll keep going and let you know if i come across any issues.

**Comment by HorizonNet** [comment]
I'm answering your pull request question here to avoid cluttering. "Coreset" is an available option during the installation of CDH. It contains the most basic components, such as HDFS, YARN, Hive, etc. Assigning the roles depends on your cluster size. In an enterprise environment you could have a cluster layout similar to the following:

1 Gateway node
3 Master nodes
x Worker nodes

The gateway would have the roles necessary to communicate with the cluster. This are definitely the gateway role of the different services or a Flume agent.
The master nodes contain all the master processes, such as the NameNode, ZooKeeper quorum etc.
The worker nodes contain all the slave processes, such as DataNodes, ResourceManager, HBase RegionServers.

**Comment by fabianrbz** [comment]
@HorizonNet thanks for the reply.
One question, I'm getting the following errors for all the databases when setting up the DBs for the cluster.
<img width="841" alt="screen shot 2017-06-14 at 17 53 09" src="https://user-images.githubusercontent.com/715229/27142153-87ae7e5a-512a-11e7-9b05-c8310ab49c0e.png">
Which is weird because i can login via the command line with  those users, passwords and databases.
what could be wrong?

**Comment by fabianrbz** [comment]
nevermind, restarted everything and it worked :man_shrugging:

**Comment by fabianrbz** [comment]
@HorizonNet do we have to update it to version 5.9 latest? By mistake I updated it to 5.11 :/, is there a problem with that version or can I still work with version 5.11? I don't think there's a possibility to downgrade...

**Comment by HorizonNet** [comment]
You should go to the latest 5.9.x version. The reason for this is that there's a minor thing you have to take care of.

**Comment by fabianrbz** [comment]
@HorizonNet any simple way of downgrading? or it would be easy to setup everything again?

**Comment by HorizonNet** [comment]
You need to downgrade the CM server and agents. Depending on your installation (f.e. with YUM) that should be quite easy. You also have to downgrade CDH. Not sure, if there is a specific order, but I would guess that you have to downgrade CDH first. The Cloudera documentation should have a section which covers downgrading of a cluster.

**Comment by fabianrbz** [comment]
@HorizonNet what's the specific reason? according to the docs 5.9 is not supported by CentoS 7.3 which is the one I'm currently using in aws. I can try to upgrade and check if it works anyway - given that I'm currently running 5.8.3 i suppose that it might work-

**Comment by fabianrbz** [comment]
you mean the step in which we need to merge the config files?

**Comment by HorizonNet** [comment]
What do you mean with "merge the config files"? Most deployments of CDH are done on RHEL 6, but if Cloudera states that 5.9 is not supported on RHEL 7.3 it doesn't mean that it doesn't work. It is just not supported in an enterprise use case.

**Comment by fabianrbz** [comment]
well as part of the upgrade process you need to merge the config files of the server if something changed.
@HorizonNet one question, as part of creating the local parcels, what's the idea? the thing is that as part of the installation process i used parcels so I already have everything in the local repo...

**Comment by HorizonNet** [comment]
During the normal installation process of CDH you're using Cloudera's remote parcel repository. The idea behind a local repository is that an enterprise cluster normally doesn't have internet access, so you need a place where you store the parcels. Also, you can roll out your own custom parcels from a local repository.

**Comment by fabianrbz** [comment]
@HorizonNet I understand the benefits of having a local repository. the thing is that I was following the instructions listed [here](https://www.cloudera.com/documentation/enterprise/latest/topics/cm_ig_create_local_parcel_repo.html#concept_cdc_kbk_mz) and when I ```cd``` to the default local repository directory -
 ```/opt/cloudera/parcel-repo.``` - everything is already there. Maybe as part of the installation process everything gets installed locally?

**Comment by HorizonNet** [comment]
Yes, that's because of your CDH installation. If you have a look at the lifecycle of a parcel, it becomes obvious that parcels get downloaded from the Cloudera parcel repository. Your task is to switch to a local repository to know what to do when you have a private environment.

**Comment by fabianrbz** [comment]
@HorizonNet yeah I know, but if they are already downloaded there is nothing much left to do right? besides creating the ones for the services...

**Comment by HorizonNet** [comment]
And again, that's not the point of this task. It doesn't make a difference if there's something already in. The idea is that you know how to set up a local repository, not how to work with it on a daily basis. That's the reason why it is such an isolated task.
